corehq/apps/fixtures/views.py

```diff
@@ -575,3 +573,83 @@ def fixture_metadata(request, domain):
     Returns list of fixtures and metadata needed for itemsets in vellum
     """
     return json_response(item_lists_by_domain(domain))
+
+
+@method_decorator(use_bootstrap5, name='dispatch')
+class CSQLFixtureExpressionView(BaseDomainView):
+    urlname = 'csql_fixture_configuration'
+    page_title = _('CSQL Fixture Confguration')
+    template_name = 'fixtures/csql_fixture_configuration.html'
+
+    @method_decorator(toggles.MODULE_BADGES.required_decorator())
+    @method_decorator(require_can_edit_fixtures)
+    def dispatch(self, request, *args, **kwargs):
+        return super(CSQLFixtureExpressionView, self).dispatch(request, *args, **kwargs)
+
+    def all_module_badge_configurations(self):
+        return CSQLFixtureExpression.by_domain(self.domain)
+
+    @property
+    def page_context(self):
+        return {
+            'save_url': reverse('csql_fixture_configuration', args=[self.domain]),
+            'csql_fixture_configurations':
+                list(self.all_module_badge_configurations().values('id', 'name', 'csql')),
+        }
+
+    def _post_response(self, message, div_class):
+        return HttpResponse((f'<div class="alert {div_class}">' + _(message) + '</div>'),
+                            content_type='text/html')
+
+    @atomic
+    def post(self, request, *args, **kwargs):
+        try:
+            data = request.POST
+            ''' Post data format is:
+                {
+                    'name': ['name1', 'name2', 'name3'],
+                    'id': ['1', '2', ''],  # empty string ID means new expression, missing means delete
+                    'csql': ['asdf', 'asdfg', 'asdfgh'],}
+                }
+            '''
+
+            touched_badge_ids = []
+            ids_list = data.getlist('id')
+            name_list = data.getlist('name')
+            csql_list = data.getlist('csql')
+
+            name_list_without_blanks = [name for name in name_list if name]
+            if len(name_list_without_blanks) != len(set(name_list_without_blanks)):
+                return self._post_response(
+                    "Configuration not updated, two expressions cannot have the same name.", 'alert-warning')
+
+            for index in range(0, len(ids_list)):
+                _id = ids_list[index]
+                name = name_list[index]
+                csql = csql_list[index]
+
+                if not name_list[index] or not csql_list[index]:
+                    if not name_list[index] and not csql_list[index]:
+                        continue  # ignore empty rows
+                    return self._post_response("Configuration not updated, some fields are blank.",
```

Comment:
I think the validation step needs to happen for all rows before the update step happens for any - otherwise you'll get a partial update if a later row fails.